code/ModelAdmin.php

```diff
@@ -416,7 +416,7 @@ protected function getManagedModelTabs()
             $forms->push(new ArrayData(array(
                 'Title' => $options['title'],
                 'Tab' => $tab,
-                'ClassName' => $options['dataClass'],
+                'ClassName' => isset($options['dataClass']) ? $options['dataClass'] : $tab,
```

Comment:
I agree with this solution

I do think there needs to at least be an inline comment explaining that this very unintuitive code (ClassName = $tab) is done for backwards compatibility, otherwise future developer will look at this with great confusion

Reply:
Added a comment